### PR TITLE
test: Code coverage :  ExpenseCardLite Component

### PR DESCRIPTION
**Pull request by CODEX108**
clickup: https://app.clickup.com/t/85zrv3vu6
image

### CODE REVIEW

File: src/app/shared/components/expense-card-lite/expense-card-lite.component.spec.ts [hunk 1]
```diff
@@ -1,24 +1,100 @@
...
+    it('should set receiptThumbnail if there is at least one thumbnail file', () => {
+      fixture.detectChanges();
+      fileService.getFilesWithThumbnail.and.returnValue(of(fileObjectData1));
+      fileService.downloadThumbnailUrl.and.returnValue(of(thumbnailUrlMockData));
+      expenseCardLiteComponent.getReceipt();
+
+      expect(fileService.getFilesWithThumbnail).toHaveBeenCalledWith(expenseCardLiteComponent.expense.id);
+      expect(fileService.getFilesWithThumbnail).toHaveBeenCalledTimes(2);
+      expect(fileService.downloadThumbnailUrl).toHaveBeenCalledWith(fileObjectData1[0].id);
+      expect(fileService.downloadThumbnailUrl).toHaveBeenCalledTimes(2);
```
Comment (Julias0) [inline src/app/shared/components/expense-card-lite/expense-card-lite.component.spec.ts:61]: Why is the call going out to `fileService.downloadThumbnailUrl` and `fileService.getFilesWithThumbnail` twice? This seems to be a bug no

File: src/app/shared/components/expense-card-lite/expense-card-lite.component.spec.ts [hunk 2]
```diff
@@ -1,24 +1,100 @@
...
+      expect(expenseCardLiteComponent.receiptThumbnail).toEqual(thumbnailUrlMockData[0].url);
+    });
+
+    it('should set receiptThumbnail to null if there are no thumbnail files', () => {
+      fixture.detectChanges();
+      fileService.getFilesWithThumbnail.and.returnValue(of([]));
+
+      expenseCardLiteComponent.getReceipt();
+      expect(fileService.getFilesWithThumbnail).toHaveBeenCalledWith(expenseCardLiteComponent.expense.id);
+      expect(fileService.getFilesWithThumbnail).toHaveBeenCalledTimes(2);
```
Comment (Julias0) [inline src/app/shared/components/expense-card-lite/expense-card-lite.component.spec.ts:71]: Same as above

**Review by Julias0 — CHANGES_REQUESTED** [review]
Added comments

File: src/app/shared/components/expense-card-lite/expense-card-lite.component.spec.ts (hunk 2 above)
Comment (CODEX108) [reply src/app/shared/components/expense-card-lite/expense-card-lite.component.spec.ts:73]: to cover the else block in the test case, I have called `getReceipt()` explicitly after setting the return value of fileService.getFilesWithThumbnail() to an empty array.

File: src/app/shared/components/expense-card-lite/expense-card-lite.component.spec.ts [hunk 3]
```diff
@@ -1,24 +1,101 @@
...
+      expect(fileService.downloadThumbnailUrl).toHaveBeenCalledOnceWith(fileObjectData1[0].id);
+      expect(expenseCardLiteComponent.receiptThumbnail).toEqual(thumbnailUrlMockData1[0].url);
+    });
+
+    it('should return an empty array if there are no thumbnail files', () => {
+      fileService.getFilesWithThumbnail.and.returnValue(of([]));
+      fixture.detectChanges();
+      expenseCardLiteComponent.getReceipt();
+      expect(fileService.getFilesWithThumbnail).toHaveBeenCalledWith(expenseCardLiteComponent.expense.id);
+      expect(fileService.getFilesWithThumbnail).toHaveBeenCalledTimes(2);
```
Comment (Julias0) [inline src/app/shared/components/expense-card-lite/expense-card-lite.component.spec.ts:73]: lets remove this

**Review by Julias0 — APPROVED** [review]
1 minor comment - fix and merge

**Review by jayfyle — APPROVED** [review]

**Pull request merged**